libdnf/rpm/transaction.cpp

```diff
@@ -472,20 +483,33 @@ class Transaction::Impl {
     CallbackInfo cb_info{nullptr, this};
     FD_t fd_in_cb{nullptr};  // file descriptor used by transaction in callback (install/reinstall package)
     std::map<unsigned int, TransactionItem *> items{};
+    bool downgrade_requested{false};
 
     /// Add package to be installed to transaction set.
     /// The transaction set is checked for duplicate package names.
     /// If found, the package with the "newest" EVR will be replaced.
     /// @param item  item to be erased
-    /// @param upgrade  operation: true - upgrade, false - install
-    void install_or_upgrade(TransactionItem & item, bool upgrade) {
+    /// @param action  one of TransactionItemAction::UPGRADE,
+    ///     TransactionItemAction::DOWNGRADE, TransactionItemAction::INSTALL
+    void install_up_down(TransactionItem & item, libdnf::transaction::TransactionItemAction action) {
+        std::string msg_action;
```

Comment:
What about to have the body somewhere else?

Reply:
Method definition has been moved out of declaration section.